packages/orchestrator-ui-components/src/components/WFOWorkflowSteps/WFOWorkflowStepList/WFOWorkflowStepList.tsx

```diff
@@ -20,36 +20,53 @@ export const WFOWorkflowStepList = React.forwardRef(
 
         const t = useTranslations('processes.steps');
 
-        const allDetailsClosedState = new Map(
-            steps.map((_, index) => [index, false]),
-        );
-        const allDetailsOpenedState = new Map(
-            steps.map((_, index) => [index, true]),
+        const initialStepListItems: StepListItem[] = steps.map((step) => ({
+            step,
+            isExpanded: false,
+        }));
+        const [stepListItems, setStepListItems] =
+            useState(initialStepListItems);
+
+        const allStepsAreExpanded = stepListItems.every(
+            (item) => item.isExpanded,
         );
 
-        const [stepDetailStates, setStepDetailStates] = useState<
-            Map<number, boolean>
-        >(new Map(allDetailsClosedState));
+        const setExpandedStateStepListItems = (isExpanded: boolean) => {
```

Comment:
Nitpicking here but does this function expand all step list items and thus wouldn't expandAllListItems be a more descriptive name?

Reply:
It would SET all isExpanded stepListItems to to given value (true/false). I would call that the expandedState. Have to admit it can be better named. "expandAllListItems" would not cover it all the way since you can also collapse everything with this function. Open for suggestions to improve this name.